packages/ui/src/components/va-select/VaSelect.demo.vue

```diff
@@ -11,6 +11,109 @@
       />
       <p>Value: {{ defaultSingleSelect.value }}</p>
     </VbCard>
+    <VbCard
+      title="Autocomplete"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="defaultSingleSelect.value"
+        class="mb-6"
+        :options="defaultSingleSelect.options"
+        autocomplete
+      />
+      <p>Value: {{ defaultSingleSelect.value }}</p>
+    </VbCard>
+    <VbCard
+      title="Multiple autocomplete"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="defaultMultiSelect.value"
+        class="mb-6"
+        :options="defaultMultiSelect.options"
+        multiple
+        autocomplete
+      />
+      <p>Value: {{ defaultMultiSelect.value }}</p>
+    </VbCard>
+    <VbCard
+      title="Multiple autocomplete with max visible options"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="defaultMultiSelect.value"
+        class="mb-6"
+        :options="defaultMultiSelect.options"
+        :max-visible-options="2"
+        multiple
+        autocomplete
+      />
+      <p>Value: {{ defaultMultiSelect.value }}</p>
+    </VbCard>
+    <VbCard
+      title="Autocomplete with create new option"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="allowCreateSelect.value"
+        class="mb-6"
+        :options="allowCreateSelect.options"
+        autocomplete
+        allow-create
+        @create-new="addNewOption"
+      />
+      <p>Value: {{ allowCreateSelect.value }}</p>
+    </VbCard>
+    <VbCard
+      title="Multiple autocomplete with content slot"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="defaultMultiSelect.value"
+        class="mb-6"
+        :options="defaultMultiSelect.options"
+        multiple
+        autocomplete
+      >
+        <template #content="{ value }">
+          <va-chip
+            v-for="chip in value"
+            :key="chip"
+            class="mr-1"
+            size="small"
+          >
+            {{ chip }}
+          </va-chip>
+        </template>
+      </va-select>
+      <p>Value: {{ defaultMultiSelect.value }}</p>
+    </VbCard>
+    <VbCard
+      title="Min chars (2) to trigger autocomplete"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="defaultSingleSelect.value"
+        class="mb-6"
+        :options="defaultSingleSelect.options"
+        :min-search-chars="2"
+        autocomplete
+      />
+      <p>Value: {{ defaultSingleSelect.value }}</p>
+    </VbCard>
+    <VbCard
+      title="Autocomplete with auto select first option"
+      style="width: 400px;"
+    >
+      <va-select
+        v-model="defaultSingleSelect.value"
+        class="mb-6"
+        :options="defaultSingleSelect.options"
+        autocomplete
+        auto-select-first-option
+      />
+      <p>Value: {{ defaultSingleSelect.value }}</p>
+    </VbCard>
     <VbCard
       title="Selected options at the top"
```

Comment:
<img width="465" alt="image" src="https://user-images.githubusercontent.com/5394573/218587075-8efb643d-b497-456b-8f54-d42cc3af0644.png">

I this case I would keep cursor at the same position it was and not jump with selected item. But please investigate, I might be mistaken. :)

Reply:
I'd prefer to move it to the separate issue, because it's not related to autocomplete and affects the whole component's logic.